express-notes-api/server.js

```diff
@@ -0,0 +1,83 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+const app = express();
+app.use(express.json());
+
+async function readData() {
+  const file = await readFile('data.json');
+  return await JSON.parse(file);
+}
+const data = await readData();
+async function dataToJson() {
+  const newObj = JSON.stringify(data, null, 2);
+  await writeFile('data.json', newObj);
+}
+
+app.get('/api/notes', (req, res) => {
+  const arr = [];
+  for (const key in data.notes) {
+    arr.push(data.notes[key]);
+  }
+  res.status(200).json(arr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const note = req.params.id;
+  if (Math.sign(note) !== 1) {
+    res.status(400).send({ error: 'id must be a positive integer' });
+  } else if (!data.notes[note]) {
+    res.status(404).send({ error: 'cannot find note with id ' + note });
+  } else {
+    res.status(200).send(data.notes[note]);
```

Comment:
Use .json here, too. I believe express automatically converts to JSON if you pass it an object, but let's be more explicit and use .json to make it clear what our intent is.